### PR TITLE
Introduce async participate functionality

### PR DESCRIPTION
**Pull request by lennartblom**

### CODE REVIEW

File: test/testParticipate.js [hunk 1]
```diff
@@ -41,6 +41,21 @@ describe("`/dabei` command", function () {
...
+    global.UrlFetchApp = {
+      fetch: function (url) {
+        assert.strictEqual(
+          url,
+          `https://slack.com/api/users.info?token=xoxb-test&user=a_user_ID`
+        );
+
+        return mockResponse;
+      },
+    };
```
Comment (lennartblom) [inline test/testParticipate.js:58]: Tests seem to be somehow broken. Adding the mocked `UrlFetchApp` helped.

File: src/main.js [hunk 2]
```diff
@@ -23,11 +76,8 @@ function doPost(e) {
 
   switch (command) {
     case PARTICIPATE_COMMAND: {
-      const result = participate(e);
-      textOutput = result[0] + listParticipants(e);
-      if (result[1]) {
-        notifyAboutParticipant(e);
-      }
+      triggerAsyncParticipate(e);
+      textOutput = "Trainingsanmeldung eingegangen :white_check_mark:";
```
Comment (lennartblom) [inline src/main.js:80]: Returning an "instant" response to avoid Slack timing out.

File: src/main.js [hunk 3]
```diff
@@ -11,6 +11,59 @@ const LIST_PARTICIPANTS_COMMAND = "/teilnehmer";
...
 
+function respondEphemeral(textOutput, responseUrl) {
+  let requestOptions = {
+    method: "post",
+    headers: {
+      "Content-type": "application/json",
+    },
+    payload: JSON.stringify({
+      text: textOutput,
+      response_type: "ephemeral"
```
Comment (lennartblom) [inline src/main.js:22]: Returning feedback of participation via ephemeral Slack message in the specific channel (only for user).

File: src/main.js [hunk 4]
```diff
@@ -11,6 +11,59 @@ const LIST_PARTICIPANTS_COMMAND = "/teilnehmer";
...
+    })
+  };
+  UrlFetchApp.fetch(responseUrl, requestOptions);
```
Comment (lennartblom) [inline src/main.js:25]: `responseUrl` is a web hook valid for ~30minutes, provided by the initial slack request data parameters.

**Review by schnika — APPROVED** [review]
nice. :)

**Pull request merged**